cli/internal/cmd/upgradecheck.go

```diff
@@ -139,42 +139,49 @@ func (u *upgradeCheckCmd) upgradeCheck(cmd *cobra.Command, fileHandler file.Hand
 	csp := conf.GetProvider()
 	u.log.Debugf("Using provider %s", csp.String())
 
-	currentServices, currentImage, currentK8s, err := u.collect.currentVersions(cmd.Context())
+	current, err := u.collect.currentVersions(cmd.Context())
 	if err != nil {
 		return err
 	}
 
-	supportedServices, supportedImages, supportedK8s, err := u.collect.supportedVersions(cmd.Context(), currentImage, csp)
+	supported, err := u.collect.supportedVersions(cmd.Context(), csp, current.image, current.k8s)
 	if err != nil {
 		return err
 	}
-	u.log.Debugf("Current service version: %s", currentServices)
-	u.log.Debugf("Supported service version: %s", supportedServices)
-	u.log.Debugf("Current k8s version: %s", currentK8s)
-	u.log.Debugf("Supported k8s version: %s", supportedK8s)
+	u.log.Debugf("Current cli version: %s", current.cli)
+	u.log.Debugf("Supported cli version(s): %s", supported.cli)
+	u.log.Debugf("Current service version: %s", current.service)
+	u.log.Debugf("Supported service version: %s", supported.service)
+	u.log.Debugf("Current k8s version: %s", current.k8s)
+	u.log.Debugf("Supported k8s version(s): %s", supported.k8s)
 
 	// Filter versions to only include upgrades
-	newServices := supportedServices
-	if err := compatibility.IsValidUpgrade(currentServices, supportedServices); err != nil {
+	newServices := supported.service
+	if err := compatibility.IsValidUpgrade(current.service, supported.service); err != nil {
 		newServices = ""
 	}
 
-	newKubernetes := filterK8sUpgrades(currentK8s, supportedK8s)
+	newKubernetes := filterK8sUpgrades(current.k8s, supported.k8s)
 	sort.Strings(newKubernetes)
```

Comment:
This should use [semver.Sort](https://pkg.go.dev/golang.org/x/mod@v0.7.0/semver#Sort) :thinking: